cmake/OpenCVCompilerOptimizations.cmake

```diff
@@ -347,11 +370,6 @@ elseif(ARM OR AARCH64)
     ocv_update(CPU_FP16_IMPLIES "NEON")
   else()
     ocv_update(CPU_KNOWN_OPTIMIZATIONS "NEON;FP16;NEON_DOTPROD;NEON_FP16;NEON_BF16")
-    ocv_update(CPU_NEON_FLAGS_ON "")
-    ocv_update(CPU_FP16_IMPLIES "NEON")
-    ocv_update(CPU_NEON_DOTPROD_IMPLIES "NEON")
-    ocv_update(CPU_NEON_FP16_IMPLIES "NEON")
-    ocv_update(CPU_NEON_BF16_IMPLIES "NEON")
```

Comment:
Why do we lost `_IMPLIES` configuration?

Reply:
I had concerns that it might affect compilation flags for dispatched or baseline files. I'll try to restore it.

Reply:
Restored.